File: find-the-distance-value-between-two-arrays-1385/solution.js
/**
 * LeetCode problem # 1385, 'Find the Distance Value Between Two Arrays'
 * https://leetcode.com/problems/find-the-distance-value-between-two-arrays
 */

/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @param {number} d
 * @return {number}
 */
function findTheDistanceValue(arr1, arr2, d) {
  arr2.sort((a, b) => a - b);
  let res = 0;

  for (let i = 0; i < arr1.length; ++i) {
    let lo = -1;
    let hi = arr2.length;

    let a = 1;

    while (lo + 1 != hi) {
      let mid = lo + ((hi - lo) >> 1);
      if (Math.abs(arr2[mid] - arr1[i]) <= d) {
        a = 0;
        break;
      }
      if (arr2[mid] > arr1[i]) {
        hi = mid;
      } else {
        lo = mid;
      }
    }
    res += a;
  }
  return res;
}

File: find-the-distance-value-between-two-arrays-1385/Solution.java

/**
 * LeetCode problem # 1385, 'Find the Distance Value Between Two Arrays'
 * https://leetcode.com/problems/find-the-distance-value-between-two-arrays/
 */

import java.util.*;

class Solution {
    public int findTheDistanceValue(int[] arr1, int[] arr2, int d) {
        Arrays.sort(arr2);
        int res = 0;

        for (int i = 0; i < arr1.length; ++i) {
            int lo = -1;
            int hi = arr2.length;

            int a = 1;

            while (lo + 1 != hi) {
                int mid = lo + (hi - lo) / 2;
                if (Math.abs(arr2[mid] - arr1[i]) <= d) {
                    a = 0;
                    break;
                }
                if (arr2[mid] > arr1[i]) {
                    hi = mid;
                } else {
                    lo = mid;
                }
            }
            res += a;
        }
        return res;
    }
}